src/colorizer/Dataset.ts

```diff
@@ -58,10 +58,32 @@ const MAX_CACHED_BACKDROPS_BYTES = 500_000_000; // 500 MB
 
 export default class Dataset {
   private frameLoader: ITextureImageLoader;
-  private frameFiles: string[];
+  private frameFiles?: string[];
   private frames: DataCache<number, Texture> | null;
   private frameDimensions: Vector2 | null;
 
+  /** Source for 3D data, resolved to http/https URLs. */
+  public frames3dSrc?: string | string[];
```

Comment:
Nit: I feel like this section could be updated to more closely match the stuff in dataset_utils

Reply:
Like a nested object?